test/tests/unit/response_test.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const expect = require('chai').expect;
+const MongoError = require('../../../lib/error').MongoError;
+const mock = require('mongodb-mock-server');
+const Server = require('../../../lib/topologies/server');
+const Long = require('bson').Long;
+
+const test = {};
+describe('Response', function() {
+  afterEach(() => mock.cleanup());
+  beforeEach(() => {
+    return mock.createServer().then(mockServer => {
+      test.server = mockServer;
+    });
+  });
+
+  it('should throw when document is error', {
+    metadata: { requires: { topology: ['single'] } },
+    test: function(done) {
+      const errdoc = {
+        ok: 0,
```

Comment:
It's nice to have the whole error response, but I think generally you _just_ need `errmsg` to trigger the code you're looking to trigger right? We should try to be least verbose in these types of mock tests IMO